### DISCUSSION TITLE
Tugboat.cs "StopClient" throws exception on start editor play if ide debugger is attached to unity.

### DISCUSSION THREAD

**Discussion by Baerikus** [FirstGearGames/FishNet#326] [Feature Request]
As the screenshot below shows, the "StopClient" function in Tugboat.cs at line 461 constantly throws a null reference exception if the IDE debugger (in this case rider) is attached to unity, and the editor play mode is started. A null check would prevent this and the need for interaction by continuing with F5 by the dev. The same could be valid for the "StopServer" function at line 431, but I did not specifically test it.

![2023_04_13_15_20_11_Sandbox_Tugboat cs_Debugger_Error](https://user-images.githubusercontent.com/83476064/231771990-217e3fda-6929-4958-81a1-6400e39090ea.png)

Changing:

`return  _client.StopConnection();`

to:

`return _client != null && _client.StopConnection();`

Fixed the problem for me.

**Comment by FirstGearGames** [comment]
Was a bug, and was resolved.
